schema/validator.go

```diff
@@ -88,11 +89,43 @@ func (v Validator) Validate(src io.Reader) error {
 
 type unimplemented string
 
-func (v unimplemented) Validate(src io.Reader) error {
+func (v unimplemented) Validate(src io.Reader, strict bool) error {
 	return fmt.Errorf("%s: unimplemented", v)
 }
 
-func validateManifestDescendants(r io.Reader) error {
+func validateMediaTypeInManifestList(r io.Reader, strict bool) error {
+	header := v1.ManifestList{}
+
+	buf, err := ioutil.ReadAll(r)
+	if err != nil {
+		return errors.Wrapf(err, "error reading the io stream")
+	}
+
+	err = json.Unmarshal(buf, &header)
+	if err != nil {
+		return errors.Wrap(err, "manifest list format mismatch")
+	}
+
+	if header.MediaType != string(v1.MediaTypeImageManifestList) {
+		msg := fmt.Sprintf("manifest list has an unknown media type: %s", header.MediaType)
+		if strict {
+			return fmt.Errorf("error: %s", msg)
```

Comment:
Agree with you and fix in https://github.com/opencontainers/image-spec/pull/452/commits/25825f3505911d69915bcfd9b2df083c23ed4e02 https://github.com/opencontainers/image-spec/pull/452/commits/7b138962e1a4b5b83da3f45c20c3f9babdc61c67 and https://github.com/opencontainers/image-spec/pull/452/commits/4daf4f71a42487d551ca41a6598d9f057d497348

I have hesitated to this, and then comply that we are usual to wrap error message. Actually I like error messages to be concise.